core/arch/x86_64/kernel/thread_private.h

```diff
@@ -50,6 +50,8 @@ struct thread_ctx {
 	struct core_mmu_user_map user_map;
 	bool have_user_map;
 	vaddr_t kern_sp;	/* Saved kernel SP during user TA execution */
+	vaddr_t svc_sp;
+	vaddr_t svc_table;
```

Comment:
how about save MSR SYSCALL_EIP in before switching to REE and restore it after switching back? then we don't need to care about others (for example, don't need to change original ldelf_handle_svc() and user_ta_handle_svc()).
same to MSR_SYSCALL_ESP.
this will help make code clean